### PR TITLE
Add Kusto cluster and database template

### PR DESCRIPTION
**Pull request by radennis**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by radennis** [comment]
Test-AzureRmResourceGroupDeployment passes, not sure why travis complains type is missing...

File: 101-kusto-cluster-database/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+      "Clusters_kustocluster_name": {
+          "type": "string",
+          "metadata": {
```
Comment (bmoore-msft) [inline 101-kusto-cluster-database/azuredeploy.json]: add a defaultValue - globally unique if needed

File: 101-kusto-cluster-database/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            "description": "Name of the cluster to create"
+          }
+      },
+      "Databases_kustodb_name": {
+          "type": "string",
```
Comment (bmoore-msft) [inline 101-kusto-cluster-database/azuredeploy.json]: add a defaultValue

File: 101-kusto-cluster-database/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      {
+          "name": "[parameters('Clusters_kustocluster_name')]",
+          "type": "Microsoft.Kusto/clusters",
+          "sku": {
+              "name": "D13_v2",
+              "tier": "Standard",
+              "capacity": 2
+          },
+          "apiVersion": "2018-09-07-preview",
+          "location": "West Europe",
```
Comment (bmoore-msft) [inline 101-kusto-cluster-database/azuredeploy.json]: Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: 101-kusto-cluster-database/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+          "tags": {
+            "Created By": "GitHub quickstart template"
+          },
+          "dependsOn": []
```
Comment (bmoore-msft) [inline 101-kusto-cluster-database/azuredeploy.json]: remove the empty array

File: 101-kusto-cluster-database/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      {
+          "name": "[concat(parameters('Clusters_kustocluster_name'), '/', parameters('Databases_kustodb_name'))]",
+          "type": "Microsoft.Kusto/clusters/databases",
+          "apiVersion": "2018-09-07-preview",
+          "location": "West Europe",
+          "properties": {
+              "softDeletePeriodInDays": 365,
+              "hotCachePeriodInDays": 31
+          },
+          "dependsOn": [
```
Comment (bmoore-msft) [inline 101-kusto-cluster-database/azuredeploy.json]: move this above properties

File: 101-kusto-cluster-database/azuredeploy.parameters.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+      "Clusters_kustocluster_name": {
+          "value": "samplecluster1234"
```
Comment (bmoore-msft) [inline 101-kusto-cluster-database/azuredeploy.parameters.json]: use GEN-UNIQUE

File: 101-kusto-cluster-database/azuredeploy.parameters.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+      "Clusters_kustocluster_name": {
+          "value": "samplecluster1234"
+      },
+      "Databases_kustodb_name": {
```
Comment (bmoore-msft) [inline 101-kusto-cluster-database/azuredeploy.parameters.json]: remove this when you have the defaultValue in the template

File: 101-kusto-cluster-database/azuredeploy.json (hunk 1 above)
Comment (radennis) [reply 101-kusto-cluster-database/azuredeploy.json]: Per the info here: https://github.com/Azure/azure-quickstart-templates/wiki/defaultValue-vs.-azuredeploy.parameters.json-vs.-GEN-UNIQUE I see it's recommended to add a GEN-UNIQUE in the properties and remove the defaultValue in azuredeploy.json therefore I did that. Let me know if I should correct anything.

**Comment by radennis** [comment]
@bmoore-msft could you please take a look?

File: 101-kusto-cluster-database/azuredeploy.json (hunk 1 above)
Comment (bmoore-msft) [reply 101-kusto-cluster-database/azuredeploy.json]: ah, ok... the page is a bit dated (I updated it)...  a defaultValue can contain an expression that is globally unique, this makes it easier for users to deploy in a "kick the tires" mode (and prod if nec).  You can use:

"defaultValue": "[concat('kusto-', uniqueString(resourceGroup().id))]"

provided a '-' is allowed in the name, if not just remove it...

File: 101-kusto-cluster-database/azuredeploy.parameters.json [hunk 8]
```diff
@@ -1,12 +1,9 @@
...
-      },
-      "Databases_kustodb_name": {
-          "value": "kustodb"
-      }
-  }
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+      "clusters_kustocluster_name": {
```
Comment (bmoore-msft) [inline 101-kusto-cluster-database/azuredeploy.parameters.json]: remove this so the defaultValue gets tested

**Comment by radennis** [comment]
@bmoore-msft friendly ping

**Pull request merged**